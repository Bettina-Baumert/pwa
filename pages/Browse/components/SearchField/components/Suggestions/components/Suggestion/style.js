/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';

export default css({
  display: 'block',
  fontSize: '1rem',
  fontWeight: 400,
  paddingLeft: 50,
  paddingRight: 50,
  paddingTop: 10,
  paddingBottom: 10,
  width: '100%',
  textAlign: 'left',
}).toString();
