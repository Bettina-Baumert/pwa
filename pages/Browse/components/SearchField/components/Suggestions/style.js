import { css } from 'glamor';
import colors from 'Styles/colors';

const bottom = value => css({
  bottom: value,
}).toString();

const wrapper = css({
  position: 'absolute',
  left: 0,
  right: 0,
  top: 124,
  backgroundColor: colors.light,
  overflowY: 'scroll',
  zIndex: 3,
  borderTop: `0.5px ${colors.dividers} solid`,
  paddingTop: 5,
}).toString();

export default {
  bottom,
  wrapper,
};
