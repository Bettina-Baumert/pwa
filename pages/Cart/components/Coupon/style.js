import cxs from 'cxs';
import { colors, variables } from 'Templates/styles';

const item = cxs({
  fontSize: '0.875rem',
  padding: `${variables.gap.small / 2}px ${variables.gap.big}px`,
});

const grid = cxs({
  flexDirection: 'column',
});

const icon = cxs({
  fontSize: '3rem',
  flexShrink: 0,
  margin: '5px 12px 0 12px',
});

const content = cxs({
  display: 'flex',
  flexDirection: 'column',
  paddingLeft: variables.gap.big,
  paddingTop: variables.gap.small,
  paddingBottom: variables.gap.small,
});

const contentLast = cxs({
  alignItems: 'flex-end',
});

const title = cxs({
  fontSize: 16,
  fontWeight: 500,
  lineHeight: 1.125,
  marginBottom: 4,
});

const savedPrice = cxs({
  fontSize: '1rem',
  fontWeight: 500,
  color: colors.primary,
});

const closeIcon = cxs({
  padding: 0,
  fontSize: 24,
  marginTop: -2,
  marginRight: -5,
});

export default {
  item,
  grid,
  savedPrice,
  icon,
  content,
  contentLast,
  title,
  closeIcon,
};
