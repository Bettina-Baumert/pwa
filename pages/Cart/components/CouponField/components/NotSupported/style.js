import { css } from 'glamor';
import variables from 'Styles/variables';
import colors from 'Styles/colors';

export default css({
  background: colors.background,
  display: 'block',
  fontSize: 12,
  margin: '20px 0 -10px',
  padding: `0 ${variables.gap.big}px`,
  textAlign: 'left',
}).toString();
