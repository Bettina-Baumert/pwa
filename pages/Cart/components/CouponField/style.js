import cxs from 'cxs';
import { colors, variables } from 'Templates/styles';

const easing = '450ms cubic-bezier(0.23, 1, 0.32, 1)';

const wrapper = cxs({
  background: colors.light,
  padding: variables.gap.big,
});

const container = cxs({
  position: 'relative',
  height: 24,
  width: '100%',
  fontSize: '0.875rem',
});

const icon = cxs({
  color: colors.primary,
  fontSize: '1.875rem',
  position: 'relative',
  float: 'right',
  top: -10,
  transition: `opacity ${easing}`,
});

const underlineWrapper = cxs({
  position: 'absolute',
  width: '100%',
  borderBottom: `1px solid ${colors.shade5}`,
  top: 24,
});

const underline = cxs({
  position: 'relative',
  width: '100%',
  top: 1,
  height: 2,
  willChange: 'transform',
  transition: `transform ${easing}`,
  background: colors.primary,
});

const underlineBlurred = cxs({
  transform: 'scale3d(0,1,1)',
});

const label = cxs({
  position: 'absolute',
  pointerEvents: 'none',
  bottom: 2,
  color: colors.shade4,
});

const input = cxs({
  position: 'absolute',
  width: 'calc(100% - 36px)',
  top: 0,
  outline: 0,
  bottom: 6,
});

export default {
  wrapper,
  container,
  icon,
  underlineWrapper,
  underline,
  underlineBlurred,
  label,
  input,
};
