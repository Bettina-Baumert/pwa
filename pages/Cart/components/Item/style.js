/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import colors from 'Styles/colors';
import variables from 'Styles/variables';

const messagesContainer = css({
  padding: variables.gap.big,
  backgroundColor: colors.light,
}).toString();

const messages = css({

  borderRadius: 4,
}).toString();

export default {
  messagesContainer,
  messages,
};
