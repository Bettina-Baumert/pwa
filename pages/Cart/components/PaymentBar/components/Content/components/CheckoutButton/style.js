import { css } from 'glamor';

const button = css({
  width: '100%',
}).toString();

const link = css({
  display: 'block',
  textAlign: 'center',
}).toString();

export default {
  button,
  link,
};
