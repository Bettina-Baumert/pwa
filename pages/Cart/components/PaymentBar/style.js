/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import variables from 'Styles/variables';

const duration = 300;
const delay = 300;
const easing = 'cubic-bezier(0.25, 0.1, 0.25, 1)';

const container = css({
  padding: `${variables.gap.big}px ${variables.gap.small}px`,
  paddingBottom: [
    `${variables.gap.big}px`,
    `calc(${variables.gap.big}px + constant(safe-area-inset-bottom))`,
    `calc(${variables.gap.big}px + env(safe-area-inset-bottom))`,
  ],
  lineHeight: 1.45,
}).toString();

const column = css({
  padding: `0 ${variables.gap.small}px`,
}).toString();

const labelColumn = css({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'flex-start',
}).toString();

const costsColumn = css({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'flex-end',
}).toString();

const buttonColumn = costsColumn;

const checkoutButton = css({
  display: 'flex',
  justifyContent: 'flex-end',
  marginTop: 1,
}).toString();

const slideInCartPaymentBarDrawer = css.keyframes({
  '0%': { transform: 'translateY(100%)' },
  '100%': { transform: 'translateY(0)' },
}).toString();

const animation = {
  in: css({
    animationName: slideInCartPaymentBarDrawer,
    animationDelay: `${delay}ms`,
    animationDuration: `${duration}ms`,
    animationTimingFunction: easing,
    animationDirection: 'normal',
    animationFillMode: 'both',
    animationIterationCount: 1,
  }).toString(),
  out: css({
    display: 'none',
  }).toString(),
};

export default {
  container,
  column,
  labelColumn,
  costsColumn,
  buttonColumn,
  checkoutButton,
  animation,
};
