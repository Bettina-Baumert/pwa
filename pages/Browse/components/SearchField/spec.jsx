/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import SearchField from './index';
import style from './style';

// Mock the redux connect() method instead of providing a fake store.
jest.mock('./connector', () => obj => obj);

/**
 * Converts a style class to a compatible selector.
 * @param {string} className The class name.
 * @returns {string} An enzyme selector for className.
 */
const getClassSelector = className => className.split(/\s+/).join('.');

jest.useFakeTimers();

describe('<SearchField />', () => {
  const defaultProps = {
    setSearchPhrase: () => {},
    submitSearch: () => {},
    searchPhrase: 'foo',
  };

  let element;

  beforeEach(() => {
    const props = {
      ...defaultProps,
      setSearchPhrase: jest.fn(),
      submitSearch: jest.fn(),
    };
    element = shallow(<SearchField {...props} />);
  });

  it('should render the search field', () => {
    expect(element).toMatchSnapshot();

    expect(element.find('Input').length).toBe(1);
    expect(element.find(`.${getClassSelector(style.button)}`).length).toBe(1);
  });

  it('should show the overlay when focused', () => {
    expect(element.find(`.${getClassSelector(style.overlay)}`).length).toBe(0);

    element.instance().handleFocusChange(true);
    jest.runAllTimers();

    const overlay = element.find(`.${getClassSelector(style.overlay)}`);
    expect(overlay.length).toBe(1);
  });

  it('should clear the', () => {
    expect(element.find(`.${getClassSelector(style.overlay)}`).length).toBe(0);

    element.instance().handleFocusChange(true);
    jest.runAllTimers();

    const overlay = element.find(`.${getClassSelector(style.overlay)}`);
    expect(overlay.length).toBe(1);
  });
});
