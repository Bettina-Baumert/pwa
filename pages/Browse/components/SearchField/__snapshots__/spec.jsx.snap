// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> should render the search field 1`] = `
<div>
  <div
    className="css-1jyozbi"
  >
    <div
      className="css-1utu657"
    >
      <form
        action="."
        onSubmit={[Function]}
      >
        <label
          className="css-y5wpat"
          htmlFor="search"
        >
          <div
            className="css-tzdtea"
          >
            <Magnifier />
          </div>
        </label>
        <Component
          autoComplete={false}
          className="css-1yoassp"
          onChange={[Function]}
          onFocusChange={[Function]}
          onSubmit={[Function]}
          setRef={[Function]}
          type="search"
          value="foo"
        />
      </form>
    </div>
    <div>
      <button
        className="css-hnvv5u css-gyrv08"
        onClick={[Function]}
      >
        <Translate
          className=""
          params={Object {}}
          string="search.cancel"
        />
      </button>
    </div>
  </div>
</div>
`;
